src/nvim/main.c

```diff
@@ -343,7 +343,7 @@ int main(int argc, char **argv)
   // Don't redraw until much later.
   RedrawingDisabled++;
 
-  setbuf(stdout, NULL);
+  setvbuf(stdout, NULL, _IONBF, BUFSIZ);
```

Comment:
although the size is formally ignored, `BUFSIZ` is a little misleading, `0` would be better. or just keep `setbuf(stdout, NULL)` as there is nothing wrong with the `setbuf()` form for this usecase.